### PR TITLE
Ers 254 submodule refactoring

### PR DESCRIPTION
**Pull request by robertkrantz**

### CODE REVIEW

File: service/src/main/java/eu/europa/ec/fisheries/ers/activity/message/consumer/bean/ActivityMessageConsumerBean.java [hunk 1]
```diff
@@ -99,11 +106,13 @@ public void onMessage(Message message) {
             }
             switch (method) {
                 case GET_FLUX_FA_REPORT:
```
Comment (RobinMilding) [inline service/src/main/java/eu/europa/ec/fisheries/ers/activity/message/consumer/bean/ActivityMessageConsumerBean.java]: It looks weird how we send "saveReportRequest" to "handleFaReportSaving", under the case labeled "GET_FLUX_FA_REPORT". Are we getting or are we saving?

File: service/src/main/java/eu/europa/ec/fisheries/ers/activity/message/consumer/bean/ActivityMessageConsumerBean.java [hunk 2]
```diff
@@ -110,7 +121,10 @@ public void onMessage(Message message) {
                     activityRulesModuleServiceBean.sendSyncAsyncFaReportToRules(faRepQueryResponseAfterMapping, "getTheOnValueFromSomewhere", getReportRequest.getRequestType(), textMessage.getJMSMessageID());
                     break;
                 case GET_FISHING_TRIPS:
-                    getFishingTripListEvent.fire(new EventMessage(textMessage));
+                    FishingTripRequest baseRequest = JAXBMarshaller.unmarshallTextMessage(textMessage, FishingTripRequest.class);
+                    FishingTripResponse baseResponse = fishingTripService.filterFishingTripsForReporting(FishingActivityRequestMapper.buildFishingActivityQueryFromRequest(baseRequest));
+                    String response = JAXBMarshaller.marshallJaxBObjectToString(baseResponse);
+                    producer.sendResponseMessageToSender(textMessage, response, 3_600_000L);
```
Comment (RobinMilding) [inline service/src/main/java/eu/europa/ec/fisheries/ers/activity/message/consumer/bean/ActivityMessageConsumerBean.java]: I guess the number is a TTL? Maybe create a constant with a good name for it?

File: service/src/test/java/eu/europa/ec/fisheries/ers/service/bean/ActivityMessageConsumerBeanTest.java [hunk 3]
```diff
@@ -146,4 +169,48 @@ public void testThrowing() {
         verify(errorEvent, times(1)).fire(any(EventMessage.class));
     }
 
+    @Test
+    public void getNonUniqueIds() throws ActivityModelMarshallException, JMSException {
```
Comment (RobinMilding) [inline service/src/test/java/eu/europa/ec/fisheries/ers/service/bean/ActivityMessageConsumerBeanTest.java]: You didn't just move the test, you improved it!

File: service/src/test/java/eu/europa/ec/fisheries/ers/service/bean/ActivityMessageConsumerBeanTest.java [hunk 4]
```diff
@@ -131,10 +125,10 @@ public void testOnMessageMethod() {
             PowerMockito.verifyStatic();
             MappedDiagnosticContext.addMessagePropertiesToThreadMappedDiagnosticContext(activeMQTextMessage);
         }
-        verify(receiveFishingActivityEvent, times(2)).fire(any(EventMessage.class));
-        verify(getFishingTripListEvent, times(1)).fire(any(EventMessage.class));
-        verify(getFACatchSummaryReportEvent, times(1)).fire(any(EventMessage.class));
-        verify(getNonUniqueIdsRequest, times(1)).fire(any(EventMessage.class));
+//        verify(receiveFishingActivityEvent, times(2)).fire(any(EventMessage.class));
```
Comment (RobinMilding) [inline service/src/test/java/eu/europa/ec/fisheries/ers/service/bean/ActivityMessageConsumerBeanTest.java]: Why leave them?

File: service/src/main/java/eu/europa/ec/fisheries/ers/activity/message/consumer/bean/ActivitySubscriptionCheckMessageConsumerBean.java [hunk 5]
```diff
@@ -59,60 +54,54 @@
...
     public void onMessage(Message message) {
-
-        TextMessage textMessage = null;
+        log.debug("Received subscription message");
+        TextMessage textMessage = (TextMessage) message;
         try {
-            textMessage = (TextMessage) message;
+            String jmsMessageID = message.getJMSMessageID();
             MappedDiagnosticContext.addMessagePropertiesToThreadMappedDiagnosticContext(textMessage);
-            ActivityModuleRequest request;
```
Comment (RobinMilding) [inline service/src/main/java/eu/europa/ec/fisheries/ers/activity/message/consumer/bean/ActivitySubscriptionCheckMessageConsumerBean.java]: Why the hell was this two different lines...

**Review by RobinMilding — APPROVED** [review]
Looks a lot better now!

File: service/src/main/java/eu/europa/ec/fisheries/ers/activity/message/consumer/bean/ActivityMessageConsumerBean.java (hunk 1 above)
Comment (robertkrantz) [reply service/src/main/java/eu/europa/ec/fisheries/ers/activity/message/consumer/bean/ActivityMessageConsumerBean.java]: It's saving a report but it's definitely a weird name, but can't really change the method types right now.

File: service/src/test/java/eu/europa/ec/fisheries/ers/activity/service/bean/ActivityMessageConsumerBeanTest.java [hunk 6]
```diff
@@ -55,20 +48,20 @@ This file is part of the Integrated Fisheries Data Management (IFDM) Suite. The
...
 
 /**
  * Created by kovian on 17/07/2017.
  */
 @RunWith(PowerMockRunner.class)
-@PrepareForTest({MappedDiagnosticContext.class})
-@PowerMockIgnore({"javax.management.*"})
+//@PrepareForTest({MappedDiagnosticContext.class})
+//@PowerMockIgnore({"javax.management.*"})
 // TODO: Rewrite this test without powermock
```
Comment (RobinMilding) [inline service/src/test/java/eu/europa/ec/fisheries/ers/activity/service/bean/ActivityMessageConsumerBeanTest.java]: You could remove all of these three lines

**Review by RobinMilding — APPROVED** [review]

File: service/src/test/java/eu/europa/ec/fisheries/ers/activity/service/bean/ActivityMessageConsumerBeanTest.java (hunk 6 above)
Comment (robertkrantz) [reply service/src/test/java/eu/europa/ec/fisheries/ers/activity/service/bean/ActivityMessageConsumerBeanTest.java]: Oops

**Review by RobinMilding — APPROVED** [review]

File: pom.xml [hunk 7]
```diff
@@ -344,6 +344,20 @@
                 <version>${uvms.maven.version}</version>
                 <type>pom</type>
                 <scope>test</scope>
+                <exclusions>
```
Comment (RobinMilding) [inline pom.xml:347]: Good thinking

**Review by RobinMilding — APPROVED** [review]

**Pull request merged**